modules/transcription-service-workflowoperation/src/main/java/org/opencastproject/transcription/workflowoperation/NibityAttachTranscriptionOperationHandler.java

```diff
@@ -121,45 +121,42 @@ public WorkflowOperationResult start(final WorkflowInstance workflowInstance, Jo
       // Get transcription result zip file from the service
       MediaPackageElement transcription = service.getGeneratedTranscription(mediaPackage.getIdentifier().compact(), jobId);
 
-      // Extract the transcript vtt
-      String captionsVtt = null;
+      String captionsZipNameVtt = mediaPackage + ".vtt";
+      String captionsZipNameDocx = mediaPackage + ".docx";
       ZipFile zipFile = new ZipFile(workspace.get(transcription.getURI()));
-      String captionsZipName = mediaPackage + ".vtt";
-      ZipEntry zippedVtt = zipFile.getEntry(captionsZipName);
-
-      if (zippedVtt != null) {
-        InputStream zis = zipFile.getInputStream(zippedVtt);
-        MediaPackageElementBuilder builder = MediaPackageElementBuilderFactory.newInstance().newElementBuilder();
-        MediaPackageElement vttElement = builder.newElement(Attachment.TYPE, new MediaPackageElementFlavor("captions", "vtt"));
-        vttElement.setIdentifier(UUID.randomUUID().toString());
-        vttElement.setMimeType(MimeType.mimeType("text", "vtt"));
-        URI vttURI = workspace.put(mediaPackage.getIdentifier().toString(), vttElement.getIdentifier(), "captions.vtt", zis);
-        vttElement.setURI(vttURI);
-        mediaPackage.add(vttElement);
-
-        // Set the target flavor if informed
-        if (flavor != null)
-          vttElement.setFlavor(flavor);
-
-        // Add tags
-        if (targetTagOption != null) {
-          for (String tag : asList(targetTagOption)) {
-            if (StringUtils.trimToNull(tag) != null)
-              vttElement.addTag(tag);
-          }
-        }
+      ZipEntry zippedVtt = zipFile.getEntry(captionsZipNameVtt);
+      ZipEntry zippedDocx = zipFile.getEntry(captionsZipNameDocx);
+
+      if (zippedVtt == null && zippedDocx == null) {
+        logger.debug("No captions found in results zip file {}", transcription.getURI());
+        throw new WorkflowOperationException("No captions found in the zip file");
```

Comment:
The debug and exception text here is not accurate; neither captions nor transcript found in the zip file is better.